synapse/handlers/e2e_keys.py

```diff
@@ -284,30 +284,7 @@ def get_cross_signing_keys_from_cache(self, query, from_user_id):
         self_signing_keys = {}
         user_signing_keys = {}
 
-        for user_id in query:
-            # XXX: consider changing the store functions to allow querying
-            # multiple users simultaneously.
-            key = yield self.store.get_e2e_cross_signing_key(
```

Comment:
the loss of a `yield` turns this from a generator function to a regular function, which means that it won't work with `defer.inlineCallbacks`.

two options for a fix:

1. remove `defer.inlineCallbacks` and change the return statement to `return defer.success({ ... })` etc
2. hackily add a `yield defer.succeed(None)` in the middle